corehq/apps/cloudcare/templates/formplayer/menu_list.html

```diff
@@ -47,6 +47,9 @@ <h3><%- displayText ? displayText : "" %></h3>
             <% } %>
           </div>
         <% } %>
+        <% if (typeof badgeText !== 'undefined' && badgeText) { %>
```

Comment:
Nitpick: if you're going to test the type, `foo === undefined` is less susceptible to typos than checking the string `"undefined"`. But in any case, why not reduce this condition to just `if (badgeText)`?

Is there a risk that badgeText could be the number `0`, or is it guaranteed to be a string?

Reply:
Yeah I had to google to find this, but I needed to check _whether the variable exists at all_, not just whether it's `undefined`, because `badgeText` is something that I'm adding at the same time on the formplayer backend, and I want this not to break anything even before upgrading formplayer. If I used `if (badgeText)` or `if (badgeText === undefined)` here, then if badgeText doesn't show up in the JSON, and thus isn't defined here at all, that'll throw an error and halt page rendering.

If this is maybe too unclear, how could I make it clearer to others. Maybe just a comment?

Reply:
Oh, I get it, that does sound familiar - I've run into issues with underscore templates erroring when variables aren't defined. It sounds like once the formplayer backend changes are released, this check could be removed, is that true? If you can delete it in a later PR that seems ideal. If you can't delete it in a later PR, a comment would help...I'd be likely to see this code in the future when I've forgotten this conversation, and try to "clean it up" by removing the `typeof badgeText !== 'undefined'`.

Reply:
I think we want to keep it around for a little while at least, but after a short overlap period it'd be fine to simplify.

Reply:
sounds good, and thanks for the comment